course7/homeworks/waleed/complex/calculator.cpp

```diff
@@ -0,0 +1,52 @@
+#include "calculator.h"
+
+#include <sstream>
+
+void Calculator::PushLine(const std::string& str)
+{
+    if (str.size() == 1) // Operator
+    {
+        if (st.size() < 2)
+        {
+            throw std::runtime_error("Not enough operands to perform operation");
+        }
+        Complex operand2 = st.top(); st.pop();
+        Complex operand1 = st.top(); st.pop();
+        Complex newTop;
+        switch (str[0])
```

Comment:
Maybe print a message in case of an unrecognized operator, so that it's clear what happened.

Reply:
Or even better: throw and exception